frontend/packages/ux-editor-v3/src/components/config/RuleComponent.tsx

```diff
@@ -220,25 +219,51 @@ class Rule extends React.Component<IRuleComponentProps, IRuleComponentState> {
           ) : null}
           <div className={classes.buttonsContainer}>
             {this.state.ruleConnection.selectedFunction ? (
-              <button onClick={this.handleSaveEdit} type='submit' className={classes.saveButton}>
+              <StudioButton
+                onClick={this.handleSaveEdit}
+                type='submit'
+                className={classes.saveButton}
+              >
                 {this.props.t('general.save')}
-              </button>
+              </StudioButton>
             ) : null}
             {this.props.connectionId ? (
-              <button
+              <StudioButton
                 type='button'
                 className={classes.dangerButton}
                 onClick={this.handleDeleteConnection}
               >
                 {this.props.t('general.delete')}
-              </button>
+              </StudioButton>
             ) : null}
-            <button className={classes.cancelButton} onClick={this.props.cancelEdit}>
+            <StudioButton
+              className={classes.cancelButton}
+              onClick={() => {
+                this.state.dialogRef.current?.close();
+              }}
+            >
               {this.props.t('general.cancel')}
-            </button>
+            </StudioButton>
           </div>
-        </div>
-      </Modal>
+        </StudioModal.Dialog>
+      </StudioModal.Root>
+    );
+  }
+
+  private renderTrigger(): React.ReactNode {
```

Comment:
Maybe we should use the `ReactElement` type instead as return type? 😄

Reply:
Agreed, it seems like `ReactElement` is a more appropriate return type here👍